tests/common-tests.h

```diff
@@ -288,8 +348,43 @@ template<> struct type_conversion<PhonebookEntry3>
     }
 };
 
+template<typename T>
+struct type_conversion<DdlTester<T>>
+{
+    typedef soci::values base_type;
+    static void from_base(soci::values const &v, soci::indicator, DdlTester<T>& t)
+    {
+        map_ddl_tester(t, v);
```

Comment:
Is there any reason not to just inline this function here, if it's only called from here?

If nothing else, this would make it more clear what it does.

Reply:
If we want to add non-arithmetic types, we will need explicit templates specialization (otherwise static_casts will not compile) and those cannot be declared static inline.

Reply:
I'd prefer to keep things simple for now and refactor later if necessary rather than making them more complicated than needed already.

And if you really want to keep this function, please find a better name for it and add a comment explaining what it does and why it can't be inlined here.

TIA!

Reply:
templates are implicitely inlined and free functions templates requires to be explicitely not static:
`warning C4499: 'static': an explicit specialization cannot have a storage class (ignored)`

Reply:
Sorry, not sure what you're doing, but are you trying to specialize `from_base()`?

What I meant was not to specialize anything at all, just use the primary template.

Reply:
Updated